batch/src/main/java/org/openmrs/analytics/JdbcFetchUtil.java

```diff
@@ -167,40 +168,37 @@ public Integer fetchMaxId(String tableName) throws SQLException {
 		return rangeMap;
 	}
 	
-	public Map<String, String> createFhirReverseMap(String searchString, String tableFhirMapPath) throws Exception {
+	public Map<String, ArrayList<String>> createFhirReverseMap(String searchString, String tableFhirMapPath)
+	        throws IOException {
 		Gson gson = new Gson();
 		Path pathToFile = Paths.get(tableFhirMapPath);
 		try (Reader reader = Files.newBufferedReader(pathToFile.toAbsolutePath(), StandardCharsets.UTF_8)) {
 			GeneralConfiguration generalConfiguration = gson.fromJson(reader, GeneralConfiguration.class);
 			Map<String, EventConfiguration> tableToFhirMap = generalConfiguration.getEventConfigurations();
 			String[] searchList = searchString.split(",");
-			Map<String, String> reverseMap = new HashMap<String, String>();
+			Map<String, ArrayList<String>> reverseMap = new HashMap<String, ArrayList<String>>();
 			for (Map.Entry<String, EventConfiguration> entry : tableToFhirMap.entrySet()) {
 				Map<String, String> linkTemplate = entry.getValue().getLinkTemplates();
 				for (String search : searchList) {
 					if (linkTemplate.containsKey("fhir") && linkTemplate.get("fhir") != null) {
 						String[] resourceName = linkTemplate.get("fhir").split("/");
+						ArrayList<String> resources = new ArrayList<String>();
 						if (resourceName.length >= 1 && resourceName[1].equals(search)) {
-							if (!reverseMap.containsKey(resourceName[1])) {
-								reverseMap.put(resourceName[1], entry.getValue().getParentTable());
+							if (reverseMap.containsKey(entry.getValue().getParentTable())) {
+								resources = reverseMap.get(entry.getValue().getParentTable());
+								resources.add(resourceName[1]);
 							} else {
-								log.error("Some tables are mapped to the same Resources");
-								throw new Exception("config file has duplicate Resource mappings");
+								resources.add(resourceName[1]);
 							}
+							reverseMap.put(entry.getValue().getParentTable(), resources);
```

Comment:
Please move this to line 191 as well (i.e., the `else` section). In the other case that the map already contains the parent table name, we don't need to create a new list and/or add it to the map. We just need to update the current list in the map.